### PR TITLE
add base jwt functionality

### PR DESCRIPTION
**Pull request by Fernando-Rz**

### CODE REVIEW

File: internal/auth/jwt/jwt.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import (
+	"errors"
+	"os"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+var jwtKey = []byte(os.Getenv("JWT_SECRET_KEY"))
```
Comment (BrayanMG) [inline internal/auth/jwt/jwt.go:11]: Does this key have to be something specific?

File: internal/http/middleware/jwt/jwt.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package jwt
```
Comment (BrayanMG) [inline internal/http/middleware/jwt/jwt.go:1]: What if we renamed this package to auth?

File: internal/auth/jwt/jwt.go (hunk 1 above)
Comment (Fernando-Rz) [reply internal/auth/jwt/jwt.go:11]: Yes, I generated a key using `openssl rand -base64 32` stored it as an environment variable.

File: internal/http/middleware/jwt/jwt.go (hunk 2 above)
Comment (Fernando-Rz) [reply internal/http/middleware/jwt/jwt.go:1]: yeah we could, It is weird to have 2 packages called jwt

File: internal/http/middleware/jwt/jwt.go
Comment (Fernando-Rz) [inline internal/http/middleware/jwt/jwt.go:1]: @BrayanMG Idk if its worth making a test for this file right now, I would need to have some protected routes and create a new server instance in the single unit test. I could add the tests for this in the User endpoints PR. Thoughts?

File: internal/auth/jwt/jwt.go (hunk 1 above)
Comment (BrayanMG) [reply internal/auth/jwt/jwt.go:11]: So would you need to send me this key so it's consistent among our machines? We'd also have to add it to our app service right?

File: internal/http/middleware/jwt/jwt.go
Comment (BrayanMG) [reply internal/http/middleware/jwt/jwt.go:1]: I think it's fine to not test it right now. Once you add the user endpoints you'll be able to run the API locally and test as much as you need.

File: internal/auth/jwt/jwt.go (hunk 1 above)
Comment (Fernando-Rz) [reply internal/auth/jwt/jwt.go:11]: yes

**Review by BrayanMG — APPROVED** [review]

**Pull request merged**